### PR TITLE
Add umask option to user module

### LINKED ISSUE #40359: Option to set UMASK with USER module.
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME

user

##### ANSIBLE VERSION

```
[root@awx ~]# ansible --version
ansible 2.6.0 (devel 6a4f3fb729) last updated 2018/05/17 15:18:06 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/ansible/lib/ansible
  executable location = /root/ansible/bin/ansible
  python version = 2.7.5 (default, Apr 11 2018, 07:36:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
[root@awx ~]# 

```

##### SUMMARY

Module parameter to set UMASK when creating a user. It looks like this would only work if the `local` parameter for the module is left at default (`local: no`) so that the useradd command is being run which has a `-K` opiton for UMASK setting. 

```
       -K, --key KEY=VALUE
           Overrides /etc/login.defs defaults (UID_MIN, UID_MAX, UMASK, PASS_MAX_DAYS and others).

           Example: -K PASS_MAX_DAYS=-1 can be used when creating system account to turn off password ageing, even though system account has
           no password at all. Multiple -K options can be specified, e.g.: -K UID_MIN=100-K UID_MAX=499
```

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes: #40359

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
user

##### ADDITIONAL INFORMATION
This has been opened previously on https://github.com/ansible/ansible/pull/71469.

### CODE REVIEW

**Comment by aminvakil** [comment]
RHEL 8.3 has the same issue as Fedora > 32 which has been mentioned in https://github.com/ansible/ansible/pull/71469#discussion_r479173242.

I guess this PR should be changed to check for `HOME_MODE` directive in `/etc/login.defs`, warn on setting `umask` option, then remove when conditions for integration test.

File: lib/ansible/modules/user.py [hunk 1]
```diff
@@ -250,6 +250,14 @@
...
         type: int
         version_added: "2.11"
+    umask:
+        description:
+            - Sets the umask of the user.
+            - Does nothing when used with other platforms.
+            - Currently supported on Linux.
+            - Does not work on Fedora versions >= 32.
+        type: str
+        version_added: "2.11"
```
Comment (mattclay) [inline lib/ansible/modules/user.py]:
```suggestion
        version_added: "2.12"
```

**Review by mattclay — COMMENTED** [review]
We're past the 2.11 cutoff, so this needs to be updated for 2.12. Tests will fail until we've created the stable-2.11 branch.

**Comment by aminvakil** [comment]
> We're past the 2.11 cutoff, so this needs to be updated for 2.12. Tests will fail until we've created the stable-2.11 branch.

Needs a CI rerun after this, as the only error now is because of missing 2.11 branch.

File: lib/ansible/modules/user.py [hunk 2]
```diff
@@ -529,6 +536,7 @@ def __init__(self, module):
         self.role = module.params['role']
         self.password_expire_max = module.params['password_expire_max']
         self.password_expire_min = module.params['password_expire_min']
+        self.umask = module.params['umask']
```
Comment (s-hertel) [inline lib/ansible/modules/user.py:540]: It looks like this won't work with `luseradd`, just `useradd`. We should check this option isn't provided with the `local` option. Looks good to merge after that :+1:

Comment (aminvakil) [reply lib/ansible/modules/user.py:540]: @s-hertel I've set this to fail when both options have values.
If this needs to change to warn that `umask` will not be set and check whenever `umask` has been set when both options have values, please tell me to change this.

**Comment by s-hertel** [comment]
@aminvakil thanks for adding this!

**Pull request merged**